### PR TITLE
Request Validator plugin add datatypes

### PR DESCRIPTION
**Pull request by tlblessing**
https://konghq.atlassian.net/browse/DOCS-1396 epic to add datatypes to plugins beyond the few template-autogenerated parameters. I'm not logging Jira subtickets because it's a lot of overhead.

Schema link:

https://github.com/Kong/kong-plugin-enterprise-request-validator

Direct review link:

https://deploy-preview-2650--kongdocs.netlify.app/hub/kong-inc/request-validator/

### CODE REVIEW

File: app/_hub/kong-inc/request-validator/index.md [hunk 1]
```diff
@@ -32,14 +31,16 @@ params:
...
-      required: true
+      required: false
       value_in_examples: '''[{"name":{"type": "string", "required": true}}]'''
-      description: Array of schema fields.
+      datatype: string
+      description: The request body schema specification.
 
     - name: allowed_content_types
-      required: false
+      required: true
```
Comment (murillopaula) [inline app/_hub/kong-inc/request-validator/index.md:40]: this is not required

at least one of these fields must be non-empty: 'body_schema', 'parameter_schema'

```bash
# http :8001/plugins name=request-validator
HTTP/1.1 400 Bad Request
Access-Control-Allow-Origin: *
Connection: keep-alive
Content-Length: 295
Content-Type: application/json; charset=utf-8
Date: Wed, 03 Mar 2021 14:18:06 GMT
Server: kong/2.2.0.0-enterprise-edition
X-Kong-Admin-Latency: 1404
X-Kong-Admin-Request-ID: 7Su9JpIpN3V4hEvv0xilLQNCNzS5lZHA
vary: Origin

{
    "code": 2,
    "fields": {
        "config": {
            "@entity": [
                "at least one of these fields must be non-empty: 'body_schema', 'parameter_schema'"
            ]
        }
    },
    "message": "schema violation (config.@entity: {\n  \"at least one of these fields must be non-empty: 'body_schema', 'parameter_schema'\"\n})",
    "name": "schema violation"
}
```

see this constraint on the schema.lua: https://github.com/Kong/kong-plugin-enterprise-request-validator/blob/master/kong/plugins/request-validator/schema.lua#L142-L143

so how do we communicate this?

File: app/_hub/kong-inc/request-validator/index.md [hunk 2]
```diff
@@ -32,14 +31,16 @@ params:
   consumer_id: true
   config:
     - name: body_schema
-      required: true
+      required: false
```
Comment (murillopaula) [inline app/_hub/kong-inc/request-validator/index.md:34]: `required = false` is not quite correct, see my other comment

File: app/_hub/kong-inc/request-validator/index.md [hunk 3]
```diff
@@ -48,24 +49,27 @@ params:
...
       default: "kong"
       value_in_examples:
+      datatype: string
       description: |
-        What validator to use. Supported values are `kong` (default) for using Kong's own schema
+        Which validator to use. Supported values are `kong` (default) for using Kong's own schema
         validator, or `draft4` for using a JSON Schema Draft 4-compliant validator.
 
     - name: parameter_schema
       required: false
```
Comment (murillopaula) [inline app/_hub/kong-inc/request-validator/index.md:58]: required = false is not quite correct, see my other comment

**Review by murillopaula — CHANGES_REQUESTED** [review]

File: app/_hub/kong-inc/request-validator/index.md (hunk 1 above)
Comment (tlblessing) [reply app/_hub/kong-inc/request-validator/index.md:40]: We'll handle it like other plugins with "semi" and more info in description 
image

File: app/_hub/kong-inc/request-validator/index.md (hunk 2 above)
Comment (tlblessing) [reply app/_hub/kong-inc/request-validator/index.md:34]: Changed to semi

File: app/_hub/kong-inc/request-validator/index.md (hunk 3 above)
Comment (tlblessing) [reply app/_hub/kong-inc/request-validator/index.md:58]: changed to semi since there is a dependency

**Review by murillopaula — APPROVED** [review]

**Review by HCloward — APPROVED** [review]

**Pull request merged**